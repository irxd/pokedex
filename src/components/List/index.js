import React from 'react';
import Grid from '@material-ui/core/Grid';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import { POKEMON_IMAGE_URL } from '../../utils/constant';

function List(props) {
  const { list, handleDetail, getPokemonId } = props;

  return (
    <Grid container justify="center" spacing={2}>
      <Grid container xs={8} spacing={2}>
        {
          list.map(pokemon => (
            <Grid item xs={3}>
              <Card onClick={() => {handleDetail(pokemon.name)}}>
                <CardActionArea>
                  <CardContent>
                    <img src={`${POKEMON_IMAGE_URL}/${pokemon.id || getPokemonId(pokemon.url)}.png`} alt={pokemon.name} height="128" width="128" />
                    <Typography gutterBottom variant="h5" component="h2">
                      {pokemon.name}
                    </Typography>
                    <Typography variant="body2" color="textSecondary" component="p">
                      #{pokemon.id || getPokemonId(pokemon.url)}
                    </Typography>
                  </CardContent>
                </CardActionArea>
              </Card>
            </Grid>
          ))
        }
      </Grid>
    </Grid>
  );
}

export default List;
