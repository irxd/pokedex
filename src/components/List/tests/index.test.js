import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import List from '../index';

const renderer = new ShallowRenderer();

const mockListData = [
  {
    name: "bulbasaur",
    url: "https://pokeapi.co/api/v2/pokemon/1/"
  },
  {
    name: "ivysaur",
    url: "https://pokeapi.co/api/v2/pokemon/2/"
  }
]

describe('<List />', () => {
  it('Render list component and should match the snapshot', () => {
    renderer.render(
        <List
          list={mockListData}
          handleDetail={() => {}}
          getPokemonId={() => {}}
        />
    );
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
