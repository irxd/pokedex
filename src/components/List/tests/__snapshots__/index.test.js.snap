// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> Render list component and should match the snapshot 1`] = `
<ForwardRef(WithStyles)
  container={true}
  justify="center"
  spacing={2}
>
  <ForwardRef(WithStyles)
    container={true}
    item={true}
    spacing={2}
    xs={8}
  >
    <ForwardRef(WithStyles)
      item={true}
      xs={3}
    >
      <ForwardRef(WithStyles)
        onClick={[Function]}
      >
        <ForwardRef(WithStyles)>
          <ForwardRef(WithStyles)>
            <img
              alt="bulbasaur"
              height="128"
              src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/undefined.png"
              width="128"
            />
            <ForwardRef(WithStyles)
              component="h2"
              gutterBottom={true}
              variant="h5"
            >
              bulbasaur
            </ForwardRef(WithStyles)>
            <ForwardRef(WithStyles)
              color="textSecondary"
              component="p"
              variant="body2"
            >
              #
            </ForwardRef(WithStyles)>
          </ForwardRef(WithStyles)>
        </ForwardRef(WithStyles)>
      </ForwardRef(WithStyles)>
    </ForwardRef(WithStyles)>
    <ForwardRef(WithStyles)
      item={true}
      xs={3}
    >
      <ForwardRef(WithStyles)
        onClick={[Function]}
      >
        <ForwardRef(WithStyles)>
          <ForwardRef(WithStyles)>
            <img
              alt="ivysaur"
              height="128"
              src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/undefined.png"
              width="128"
            />
            <ForwardRef(WithStyles)
              component="h2"
              gutterBottom={true}
              variant="h5"
            >
              ivysaur
            </ForwardRef(WithStyles)>
            <ForwardRef(WithStyles)
              color="textSecondary"
              component="p"
              variant="body2"
            >
              #
            </ForwardRef(WithStyles)>
          </ForwardRef(WithStyles)>
        </ForwardRef(WithStyles)>
      </ForwardRef(WithStyles)>
    </ForwardRef(WithStyles)>
  </ForwardRef(WithStyles)>
</ForwardRef(WithStyles)>
`;
